src/main/java/CustomThreadImplementingCallable.java

```diff
@@ -0,0 +1,17 @@
+import java.util.List;
+import java.util.concurrent.Callable;
+
+public class CustomThreadImplementingCallable implements Callable<Integer> {
```

Comment:
```suggestion
public class ListSumCalculator implements Callable<Integer> {
```
it's better to name classes according to their purpose (same for some of your other classes)